native/src/main/java/io/ballerina/stdlib/email/util/EmailAccessUtil.java

```diff
@@ -351,9 +355,22 @@ private static Object getJsonContent(String messageContent) {
 
     private static String extractBodyFromMessage(Message message) throws MessagingException, IOException {
         String messageBody = "";
-        if (message.getContentType() != null && CommonUtil.isTextBased(message.getContentType())) {
+        String contentType = message.getContentType();
+        if (contentType != null && CommonUtil.isTextBased(contentType.toLowerCase(Locale.getDefault()))) {
             if (message.getContent() != null) {
-                messageBody = message.getContent().toString();
+                Object content = message.getContent();
+                if (content instanceof InputStream) {
+                    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+                    int nRead;
+                    byte[] data = new byte[1024];
+                    while ((nRead = ((InputStream) content).read(data, 0, data.length)) != -1) {
```

Comment:
Where do we close this InputStream?

Reply:
Try using `try with resource` feature

Reply:
Fixed with following commits:
[d3847a2ce4fc78cdea9fd2f31983703007bf25ea](https://github.com/ballerina-platform/module-ballerina-email/pull/468/commits/d3847a2ce4fc78cdea9fd2f31983703007bf25ea)  
[dc04c69ab95635a8f0bdcf0a9745e13827eec37e](https://github.com/ballerina-platform/module-ballerina-email/pull/468/commits/dc04c69ab95635a8f0bdcf0a9745e13827eec37e)